cbioportal_etl/main.py

```diff
@@ -0,0 +1,277 @@
+    parser.add_argument("-r", "--ref-dir", required=False, help="Reference directory. Defaults to tool's ref dir if not provided.")
+    parser.add_argument("-a", "--all", required=False, action="store_true", help="Include all relevant files, not just status=active files, NOT RECOMMENDED")
+
+    # Arguments for Step 2 - diff_studies.py
+    parser.add_argument("-u", "--url", default="https://pedcbioportal.kidsfirstdrc.org/api/v2/api-docs", help="URL to search against")
+    parser.add_argument("-s", "--study", required=True, help="Cancer study ID")
+    parser.add_argument("-at", "--token", required=False, help="Token file obtained from Web API. Required if running Step 2")
+    parser.add_argument("-ds", "--datasheet-sample", default=os.path.join(os.getcwd(), "datasheets/data_clinical_sample.txt"), help="File containing cBio-formatted sample metadata (default: datasheets/data_clinical_sample.txt from Step 1 output)")
+    parser.add_argument("-dp", "--datasheet-patient", default=os.path.join(os.getcwd(), "datasheets/data_clinical_patient.txt"), help="File containing cBio-formatted patient metadata (default: datasheets/data_clinical_patient.txt from Step 1 output)")
+
+    # Arguments for Step 3 - get_files_from_manifest.py
+    parser.add_argument("-m", "--manifest", default="cbio_file_name_id.txt", help="Manifest file (default: cbio_file_name_id.txt from Step 1 output)")
+    parser.add_argument("-f", "--file-types", required=False, help="Comma-separated file types to download")
+    parser.add_argument("-t", "--aws-tbl", required=False, help="AWS table with bucket name and keys")
+    parser.add_argument("-sp", "--sbg-profile", required=False, help="SBG profile name")
+    parser.add_argument("-c", "--cbio-manifest", required=False, help="cBio manifest to limit downloads")
+    parser.add_argument("-ao", "--active-only", required=False, action="store_true", help="Only include active files")
+    parser.add_argument("-rm", "--rm-na", required=False, action="store_true", help="Remove entries where file_id and s3_path are NA")
+    parser.add_argument("-d", "--debug", required=False, action="store_true", help="Enable debug mode")
+    parser.add_argument("-o", "--overwrite", required=False, action="store_true", help="Overwrite files if they already exist")
+
+    # Arguments for Step 5 - genomics_file_cbio_package_build.py
+    parser.add_argument("-dpc", "--data-processing-config", required=False, help="Data processing configuration file. Default: value inputted for --study + '_data_processing_config.json'")
+    parser.add_argument("-sf", "--study-flag", required=True, choices=["both", "kf", "dgd"], help="Study flag")
+    parser.add_argument("-l", "--legacy", required=False, action="store_true", help="Enable legacy mode")
+
+    return parser
+
+def execute_step(step_number, args, steps):
+    step = steps.get(step_number)
+    if not step:
+        sys.stderr.write(f"Step {step_number} not defined in the pipeline.\n")
+        return
```

Comment:
Safe but unnecessary. You already stop something like this from happening by only allowing certain choices for step number arguments and hardcoding the steps.

Reply:
If you want to keep it, move it out of the function.